From f2f65f2e1b178a4b52c4b4d03acf448771b9e3fe Mon Sep 17 00:00:00 2001
From: Lionel Debieve <lionel.debieve@foss.st.com>
Date: Thu, 18 Nov 2021 14:21:39 +0100
Subject: [PATCH 0306/1695] plat-stm32mp1: add global defines in
 platform_config.h

Add the global defines to access all platform registers.
Include part and revision numbers.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 20ab1164e285d407b38d5c74a6130982abd15dd4 plat-stm32mp1: add global defines in platform_config.h]
Signed-off-by: Lionel Debieve <lionel.debieve@foss.st.com>
Change-Id: I666bc301dde99120b5a10453063f903010199703
---
 core/arch/arm/plat-stm32mp1/platform_config.h | 104 +++++++++++++++++-
 1 file changed, 102 insertions(+), 2 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp1/platform_config.h b/core/arch/arm/plat-stm32mp1/platform_config.h
index ea13784ae..772068e6b 100644
--- a/core/arch/arm/plat-stm32mp1/platform_config.h
+++ b/core/arch/arm/plat-stm32mp1/platform_config.h
@@ -38,6 +38,8 @@
 #define APB6_SIZE			0x0000d000
 #endif
 
+#define AHB2_BASE			0x48000000
+#define AHB2_SIZE			0x01040000
 #define AHB4_BASE			0x50000000
 #define AHB4_SIZE			0x00020000
 #ifdef CFG_STM32MP13
@@ -50,10 +52,20 @@
 #endif
 
 /* SoC interface registers base address */
+#define ADC1_BASE			0x48003000
+#define ADC2_BASE			0x48004000
+#define BKPSRAM_BASE			0x54000000
 #define BSEC_BASE			0x5c005000
-#define ETZPC_BASE			0x5c007000
 #define CRYP1_BASE			0x54001000
+#define DBGMCU_BASE			0x50081000
+#define DCMIPP_BASE			0x5a000000
 #define DDR_BASE			0xc0000000ul
+#define DDRCTRL_BASE			0x5a003000
+#define DDRPHYC_BASE			0x5a004000
+#define ETH1_BASE			0x5800A000
+#define ETH2_BASE			0x5800E000
+#define ETZPC_BASE			0x5c007000
+#define FMC_BASE			0x58002000
 #define GIC_BASE			0xa0021000ul
 #define GPIOA_BASE			0x50002000
 #define GPIOB_BASE			0x50003000
@@ -64,6 +76,13 @@
 #define GPIOG_BASE			0x50008000
 #define GPIOH_BASE			0x50009000
 #define GPIOI_BASE			0x5000a000
+#ifdef CFG_STM32MP13
+#define HASH1_BASE			0x54003000
+#define I2C3_BASE			0x4c004000
+#define I2C4_BASE			0x4c005000
+#define I2C5_BASE			0x4c006000
+#endif
+#ifdef CFG_STM32MP15
 #define GPIOJ_BASE			0x5000b000
 #define GPIOK_BASE			0x5000c000
 #define GPIOZ_BASE			0x54004000
@@ -71,12 +90,17 @@
 #define I2C4_BASE			0x5c002000
 #define I2C5_BASE			0x40015000
 #define I2C6_BASE			0x5c009000
+#endif
 #define IWDG1_BASE			0x5c003000
 #define IWDG2_BASE			0x5a002000
 #define LTDC_BASE			0x5a001000
+#define MCUSRAM_BASE			0x30000000ul
 #define MDMA_BASE			0x58000000
+#define OTG_BASE			0x49000000
 #define PWR_BASE			0x50001000
+#define QSPI_BASE			0x58003000
 #define RCC_BASE			0x50000000
+#define RETRAM_BASE			0x38000000ul
 #ifdef CFG_STM32MP13
 #define RNG1_BASE			0x54004000
 #endif
@@ -84,8 +108,8 @@
 #define RNG1_BASE			0x54003000
 #endif
 #define RTC_BASE			0x5c004000
-#define SPI6_BASE			0x5c001000
 #ifdef CFG_STM32MP15
+#define SPI6_BASE			0x5c001000
 #define SRAM1_BASE			0x30000000
 #define SRAM2_BASE			0x30020000
 #define SRAM3_BASE			0x30040000
@@ -100,6 +124,33 @@
 #define SYSRAM_BASE			0x2ffc0000
 #endif
 #define TAMP_BASE			0x5c00a000
+#define TIM1_BASE			0x44000000
+#define TIM2_BASE			0x40000000
+#define TIM3_BASE			0x40001000
+#define TIM4_BASE			0x40002000
+#define TIM5_BASE			0x40003000
+#define TIM6_BASE			0x40004000
+#define TIM7_BASE			0x40005000
+#define TIM8_BASE			0x44001000
+#ifdef CFG_STM32MP13
+#define LPTIM2_BASE			0x50021000
+#define LPTIM3_BASE			0x50022000
+#define TIM12_BASE			0x4c007000
+#define TIM13_BASE			0x4c008000
+#define TIM14_BASE			0x4c009000
+#define TIM15_BASE			0x4c00a000
+#define TIM16_BASE			0x4c00b000
+#define TIM17_BASE			0x4c00c000
+#endif
+#ifdef CFG_STM32MP15
+#define TIM12_BASE			0x40006000
+#define TIM13_BASE			0x40007000
+#define TIM14_BASE			0x40008000
+#define TIM15_BASE			0x44006000
+#define TIM16_BASE			0x44007000
+#define TIM17_BASE			0x44008000
+#endif
+#define TSC_BASE			0x5000B000
 #define TZC_BASE			0x5c006000
 #ifdef CFG_STM32MP13
 #define UART1_BASE			0x4c000000
@@ -115,6 +166,7 @@
 #define UART6_BASE			0x44003000
 #define UART7_BASE			0x40018000
 #define UART8_BASE			0x40019000
+#define VREFBUF_BASE			0x50025000
 
 #define ROM_BASE			0
 #define ROM_SIZE			0x20000
@@ -233,4 +285,52 @@
 #define SRAM4_SIZE			0x10000
 #endif
 
+/* RETRAM layout */
+#define RETRAM_SIZE			U(0x10000)
+
+/* MCUSRAM layout */
+#define MCUSRAM_SIZE			U(0x60000)
+
+/* SoC part numbers and revisions */
+#define STM32MP135C_PART_NB		U(0x05010000)
+#define STM32MP135A_PART_NB		U(0x05010001)
+#define STM32MP133C_PART_NB		U(0x050100C0)
+#define STM32MP133A_PART_NB		U(0x050100C1)
+#define STM32MP131C_PART_NB		U(0x050106C8)
+#define STM32MP131A_PART_NB		U(0x050106C9)
+#define STM32MP135F_PART_NB		U(0x05010800)
+#define STM32MP135D_PART_NB		U(0x05010801)
+#define STM32MP133F_PART_NB		U(0x050108C0)
+#define STM32MP133D_PART_NB		U(0x050108C1)
+#define STM32MP131F_PART_NB		U(0x05010EC8)
+#define STM32MP131D_PART_NB		U(0x05010EC9)
+
+#define STM32MP157C_PART_NB		U(0x05000000)
+#define STM32MP157A_PART_NB		U(0x05000001)
+#define STM32MP153C_PART_NB		U(0x05000024)
+#define STM32MP153A_PART_NB		U(0x05000025)
+#define STM32MP151C_PART_NB		U(0x0500002E)
+#define STM32MP151A_PART_NB		U(0x0500002F)
+#define STM32MP157F_PART_NB		U(0x05000080)
+#define STM32MP157D_PART_NB		U(0x05000081)
+#define STM32MP153F_PART_NB		U(0x050000A4)
+#define STM32MP153D_PART_NB		U(0x050000A5)
+#define STM32MP151F_PART_NB		U(0x050000AE)
+#define STM32MP151D_PART_NB		U(0x050000AF)
+
+#ifdef CFG_STM32MP13
+#define STM32MP1_CHIP_ID		U(0x501)
+#define STM32MP1_REV_A			U(0x00001000)
+#define STM32MP1_REV_Z			U(0x00001001)
+#endif
+
+#ifdef CFG_STM32MP15
+#define STM32MP1_CHIP_ID		U(0x500)
+#define STM32MP1_REV_A			U(0x00001000)
+#define STM32MP1_REV_B			U(0x00002000)
+#define STM32MP1_REV_Z			U(0x00002001)
+#endif
+
+#define STM32MP1_CHIP_DEFAULT_VERSION	U(0)
+
 #endif /*PLATFORM_CONFIG_H*/
-- 
2.47.2

